unified-runtime/source/adapters/native_cpu/enqueue.cpp

```diff
@@ -639,13 +639,25 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMFill2D(
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMMemcpy2D(
-    ur_queue_handle_t /*hQueue*/, bool /*blocking*/, void * /*pDst*/,
-    size_t /*dstPitch*/, const void * /*pSrc*/, size_t /*srcPitch*/,
-    size_t /*width*/, size_t /*height*/, uint32_t /*numEventsInWaitList*/,
-    const ur_event_handle_t * /*phEventWaitList*/,
-    ur_event_handle_t * /*phEvent*/) {
-
-  DIE_NO_IMPLEMENTATION;
+    ur_queue_handle_t hQueue, bool blocking, void *pDst, size_t dstPitch,
+    const void *pSrc, size_t srcPitch, size_t width, size_t height,
+    uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
+    ur_event_handle_t *phEvent) {
+  return withTimingEvent(
+      UR_COMMAND_USM_MEMCPY_2D, hQueue, numEventsInWaitList, phEventWaitList,
+      phEvent,
+      [width, height, srcPitch, dstPitch, pDst, pSrc]() {
+        for (size_t w = 0; w < width; w++)
+          for (size_t h = 0; h < height; h++) {
```

Comment:
This inner loop can be done as a call to `memcpy`.

Is this function likely to be called with `srcPitch == 0 && dstPitch == 0`? If so, that could also be worth special casing to turn the whole thing into one single call to `memcpy`.